scripts/scil_surface_from_labels.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Script to convert segmented volume to a surface with marching cube'
+
+Example : use wmparc.a2009s.nii.gz with some aseg.stats indices
+
+scil_surface_from_labels.py out_surface.vtk \\
+    --in_labels s1a1/mask/S1-A1_wmparc.a2009s.nii.gz\\
+    --indices 16:32 --vox2vtk --opening 2 --smooth 2 -v
+"""
+
+import argparse
+import logging
+
+import nibabel as nib
+import trimeshpy
+import trimeshpy.vtk_util as vtk_u
+from scipy.ndimage import (binary_closing,
+                           binary_dilation,
+                           binary_erosion,
+                           binary_opening,
+                           binary_fill_holes)
+import mcubes
+
+from scilpy.image.labels import get_data_as_labels
+from scilpy.io.image import get_data_as_mask, merge_labels_into_mask
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_verbose_arg,
+                             assert_inputs_exist,
+                             assert_outputs_exist,
+                             ranged_type)
+
+EPILOG = """
+References:
+[1] St-Onge, E., Daducci, A., Girard, G. and Descoteaux, M. 2018.
+    Surface-enhanced tractography (SET). NeuroImage.
+"""
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__, epilog=EPILOG,
+                                formatter_class=argparse.RawTextHelpFormatter)
+
+    g = p.add_argument_group("Input (Labels or Mask)")
+    mxg = g.add_mutually_exclusive_group(required=True)
+    mxg.add_argument('--in_labels',
+                     help='Path of the atlas (nii or nii.gz).')
+    mxg.add_argument('--in_mask',
+                     help='Path of the mask (nii or nii.gz).')
+
+    p.add_argument('out_surface',
+                   help='Output surface (.vtk)')
+
+    p.add_argument('--indices', nargs='+',
+                   help='List of labels indices to use for the surface.')
+
+    p.add_argument('--value', default=0.5,
+                   type=ranged_type(float, 0, None, min_excluded=False),
+                   help='Isosurface threshold value used. '
+                        'This value is called isovalue '
+                        'in mbcube. [%(default)s]')
```

Comment:
No :D @StongeEtienne ?

Reply:
It is the "threshold" value: the isosurface = level set.
For a binary mask (with 0 and 1) , 0.5 will generate a surface in the middle of the transition.
https://github.com/pmneila/PyMCubes/tree/master

However, it "could" be used to create a surface at any "level-set" for a volume with float values.

Currently in the script, since only "binary" format is given, 0.5 could be fixed.
(this was not the behavior in my previous script)